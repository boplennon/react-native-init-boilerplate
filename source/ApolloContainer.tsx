import { ApolloClient, HttpLink, InMemoryCache } from 'apollo-boost';
import React, { Component } from 'react';
import { ApolloProvider } from 'react-apollo';
import ReactNavigator, { appUri } from './ReactNavigator';
import { Provider as PaperProvider } from 'react-native-paper';

const client = new ApolloClient({
  cache: new InMemoryCache(),
  link: new HttpLink({
    uri: 'https://api.github.com/graphql',
  }),
});

export default class ApolloContainer extends Component {
  render() {
    return (
      <ApolloProvider client={client}>
        <PaperProvider>
          <ReactNavigator uriPrefix={appUri} />
        </PaperProvider>
      </ApolloProvider>
    );
  }
}
