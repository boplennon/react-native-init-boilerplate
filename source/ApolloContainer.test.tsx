import React from 'react';
import { shallow } from 'enzyme';

import ApolloContainer from './ApolloContainer';


function setup() {
  return shallow(<ApolloContainer />);
}

describe('components', () => {
  describe('ApolloContainer', () => {
    it('renders correctly with defaults', () => {
      const wrapper = setup();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
