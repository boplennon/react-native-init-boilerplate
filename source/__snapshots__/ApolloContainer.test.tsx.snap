// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components ApolloContainer renders correctly with defaults 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ApolloContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <UNDEFINED>
        <UNDEFINED
          uriPrefix="myapp://"
        />
      </UNDEFINED>,
      "client": ApolloClient {
        "cache": Object {
          "diff": [Function],
          "evict": [Function],
          "performTransaction": [Function],
          "read": [Function],
          "readFragment": [Function],
          "readQuery": [Function],
          "recordOptimisticTransaction": [Function],
          "removeOptimistic": [Function],
          "reset": [Function],
          "transformDocument": [Function],
          "watch": [Function],
          "write": [Function],
          "writeFragment": [Function],
          "writeQuery": [Function],
        },
        "defaultOptions": Object {},
        "disableNetworkFetches": false,
        "link": ApolloLink {
          "request": [Function],
        },
        "mutate": [Function],
        "query": [Function],
        "queryDeduplication": true,
        "reFetchObservableQueries": [Function],
        "resetStore": [Function],
        "resetStoreCallbacks": Array [],
        "ssrMode": false,
        "store": DataStore {
          "cache": Object {
            "diff": [Function],
            "evict": [Function],
            "performTransaction": [Function],
            "read": [Function],
            "readFragment": [Function],
            "readQuery": [Function],
            "recordOptimisticTransaction": [Function],
            "removeOptimistic": [Function],
            "reset": [Function],
            "transformDocument": [Function],
            "watch": [Function],
            "write": [Function],
            "writeFragment": [Function],
            "writeQuery": [Function],
          },
        },
        "version": "2.4.2",
        "watchQuery": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <UNDEFINED
          uriPrefix="myapp://"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "uriPrefix": "myapp://",
        },
        "ref": null,
        "rendered": null,
        "type": undefined,
      },
      "type": undefined,
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <UNDEFINED>
          <UNDEFINED
            uriPrefix="myapp://"
          />
        </UNDEFINED>,
        "client": ApolloClient {
          "cache": Object {
            "diff": [Function],
            "evict": [Function],
            "performTransaction": [Function],
            "read": [Function],
            "readFragment": [Function],
            "readQuery": [Function],
            "recordOptimisticTransaction": [Function],
            "removeOptimistic": [Function],
            "reset": [Function],
            "transformDocument": [Function],
            "watch": [Function],
            "write": [Function],
            "writeFragment": [Function],
            "writeQuery": [Function],
          },
          "defaultOptions": Object {},
          "disableNetworkFetches": false,
          "link": ApolloLink {
            "request": [Function],
          },
          "mutate": [Function],
          "query": [Function],
          "queryDeduplication": true,
          "reFetchObservableQueries": [Function],
          "resetStore": [Function],
          "resetStoreCallbacks": Array [],
          "ssrMode": false,
          "store": DataStore {
            "cache": Object {
              "diff": [Function],
              "evict": [Function],
              "performTransaction": [Function],
              "read": [Function],
              "readFragment": [Function],
              "readQuery": [Function],
              "recordOptimisticTransaction": [Function],
              "removeOptimistic": [Function],
              "reset": [Function],
              "transformDocument": [Function],
              "watch": [Function],
              "write": [Function],
              "writeFragment": [Function],
              "writeQuery": [Function],
            },
          },
          "version": "2.4.2",
          "watchQuery": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <UNDEFINED
            uriPrefix="myapp://"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "uriPrefix": "myapp://",
          },
          "ref": null,
          "rendered": null,
          "type": undefined,
        },
        "type": undefined,
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
